### PR TITLE
[chore] Cleanup and Refactor Proto

### PR DESCRIPTION
**Pull request by warittornc**
## Implementation details
- Do 1
- Do 2

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the GitHub PR explorer)

### CODE REVIEW

File: bothan-api/server/src/api/crypto.rs [hunk 1]
```diff
@@ -8,30 +8,34 @@ use bothan_core::manager::crypto_asset_info::error::{PushMonitoringRecordError,
...
+    PushMonitoringRecordsRequest, PushMonitoringRecordsResponse, UpdateRegistryRequest,
+    UpdateRegistryResponse,
+};
 
 pub const PRECISION: u32 = 9;
 
 /// The `CryptoQueryServer` struct represents a server for querying cryptocurrency prices.
 pub struct CryptoQueryServer {
-    manager: Arc<CryptoAssetInfoManager<'static>>,
+    crypto_manager: Arc<CryptoAssetInfoManager<'static>>,
```
Comment (colmazia) [inline bothan-api/server/src/api/crypto.rs:21]: don't change file name 'crypto' to something else ror?

Comment (colmazia) [reply bothan-api/server/src/api/crypto.rs:21]: why keep crypto na?

File: proto/bothan/v1/bothan.proto [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+package bothan.v1;
+
+import "google/api/annotations.proto";
+
+option go_package = "github.com/bothan/bothan-api/client/proto";
+
+// SignalService defines the gRPC service responsible for updating the registry
+// and managing active signal IDs.
+service BothanService {
```
Comment (colmazia) [inline proto/bothan/v1/bothan.proto:11]: comment doesn't match

File: proto/bothan/v1/bothan.proto [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  // Gets the information regarding the signal service
+  rpc GetInfo(GetInfoRequest) returns (GetInfoResponse) {
+    option (google.api.http).get = "/info";
+  }
+
+  // Updates the registry with the given IPFS hash and version.
+  // The registry stores metadata and configuration data that can be referenced
+  // by other parts of the system.
+  rpc UpdateRegistry(UpdateRegistryRequest) returns (UpdateRegistryResponse) {
```
Comment (colmazia) [inline proto/bothan/v1/bothan.proto:20]: also please keep the comment in the format of `UpdateRegistry is ...` or `UpdateRegistry does ...`

File: bothan-api/server/src/api/crypto.rs (hunk 1 above)
Comment (warittornc) [reply bothan-api/server/src/api/crypto.rs:21]: will rename server and file name but will keep internal as crypto_manager

File: proto/bothan/v1/bothan.proto (hunk 3 above)
Comment (warittornc) [reply proto/bothan/v1/bothan.proto:17]: formatted to  `Struct defines ...`

**Review by colmazia — APPROVED** [review]

File: proto/bothan/v1/bothan.proto [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+// UpdateRegistryRequest defines the request message for the UpdateRegistry RPC method.
+message UpdateRegistryRequest {
+  // The IPFS hash pointing to the registry data.
+  string ipfs_hash = 1;
+  // The version of the registry.
+  string version = 2;
+}
+
+message UpdateRegistryResponse {}
```
Comment (RogerKSI) [inline proto/bothan/v1/bothan.proto:53]: missing comment.

File: proto/bothan/v1/bothan.proto [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    option (google.api.http).post = "/monitoring_records";
+  }
+
+  // GetPrices gets prices for the specified signal IDs.
+  rpc GetPrices(GetPricesRequest) returns (GetPricesResponse) {
+    option (google.api.http).get = "/prices/{signal_ids}";
+  }
+}
+
+message GetInfoRequest {}
```
Comment (RogerKSI) [inline proto/bothan/v1/bothan.proto:33]: missing comment.

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**